stubs/D3DShot/d3dshot/dll/d3d.pyi

```diff
@@ -0,0 +1,228 @@
+class D3D11_BOX(Structure):
+    left: UINT
+    top: UINT
+    front: UINT
+    right: UINT
+    bottom: UINT
+    back: UINT
+
+class D3D11_TEXTURE2D_DESC(Structure):
+    Width: UINT
+    Height: UINT
+    MipLevels: UINT
+    ArraySize: UINT
+    Format: UINT
+    SampleDesc: DXGI_SAMPLE_DESC
+    Usage: UINT
+    BindFlags: UINT
+    CPUAccessFlags: UINT
+    MiscFlags: UINT
+
+class ID3D11DeviceChild(_IUnknown):
+    def GetDevice(self) -> None: ...
+    def GetPrivateData(self) -> _HRESULT: ...
+    def SetPrivateData(self) -> _HRESULT: ...
+    def SetPrivateDataInterface(self) -> _HRESULT: ...
+
+class ID3D11Resource(ID3D11DeviceChild):
+    def GetType(self) -> None: ...
+    def SetEvictionPriority(self) -> None: ...
+    def GetEvictionPriority(self) -> UINT: ...
+
+class ID3D11Texture2D(ID3D11Resource):
+    def GetDesc(self, pDesc: _Pointer[D3D11_TEXTURE2D_DESC]) -> None: ...
+
+class ID3D11DeviceContext(ID3D11DeviceChild):
+    def VSSetConstantBuffers(self) -> None: ...
+    def PSSetShaderResources(self) -> None: ...
+    def PSSetShader(self) -> None: ...
+    def PSSetSamplers(self) -> None: ...
+    def VSSetShader(self) -> None: ...
+    def DrawIndexed(self) -> None: ...
+    def Draw(self) -> None: ...
+    def Map(self) -> _HRESULT: ...
+    def Unmap(self) -> None: ...
+    def PSSetConstantBuffers(self) -> None: ...
+    def IASetInputLayout(self) -> None: ...
+    def IASetVertexBuffers(self) -> None: ...
+    def IASetIndexBuffer(self) -> None: ...
+    def DrawIndexedInstanced(self) -> None: ...
+    def DrawInstanced(self) -> None: ...
+    def GSSetConstantBuffers(self) -> None: ...
+    def GSSetShader(self) -> None: ...
+    def IASetPrimitiveTopology(self) -> None: ...
+    def VSSetShaderResources(self) -> None: ...
+    def VSSetSamplers(self) -> None: ...
+    def Begin(self) -> None: ...
+    def End(self) -> None: ...
+    def GetData(self) -> _HRESULT: ...
+    def SetPredication(self) -> None: ...
+    def GSSetShaderResources(self) -> None: ...
+    def GSSetSamplers(self) -> None: ...
+    def OMSetRenderTargets(self) -> None: ...
+    def OMSetRenderTargetsAndUnorderedAccessViews(self) -> None: ...
+    def OMSetBlendState(self) -> None: ...
+    def OMSetDepthStencilState(self) -> None: ...
+    def SOSetTargets(self) -> None: ...
+    def DrawAuto(self) -> None: ...
+    def DrawIndexedInstancedIndirect(self) -> None: ...
+    def DrawInstancedIndirect(self) -> None: ...
+    def Dispatch(self) -> None: ...
+    def DispatchIndirect(self) -> None: ...
+    def RSSetState(self) -> None: ...
+    def RSSetViewports(self) -> None: ...
+    def RSSetScissorRects(self) -> None: ...
+    def CopySubresourceRegion(
+        self,
+        pDstResource: _Pointer[ID3D11Resource],
+        DstSubresource: UINT,
+        DstX: UINT,
+        DstY: UINT,
+        DstZ: UINT,
+        pSrcResource: _Pointer[ID3D11Resource],
+        SrcSubresource: UINT,
+        pSrcBox: _Pointer[D3D11_BOX],
+    ) -> None: ...
```

Comment:
I don't know anything about ctypes, really. It looks like these methods are all autogenerated using some ctypes metaclass magic or something.

Are these parameters definitely positional-or-keyword? Where do the parameter names come from? https://github.com/SerpentAI/D3DShot/blob/4472120a21a1cb397d4f30d7a36af993e7544bc9/d3dshot/dll/d3d.py#L108-L120

Reply:
Parameter names all come from Microsoft's official docs. For this specific example: https://docs.microsoft.com/en-us/windows/win32/api/d3d11/nf-d3d11-id3d11devicecontext-copysubresourceregion

Reply:
Okay, great. What about the other point, though -- are these parameters definitely positional-or-keyword? I.e., is it okay to do this?

```python
from d3dshot.dll.d3d import ID3D11DeviceContext
x = ID3D11DeviceContext()
x.CopyResource(pDstResource=<whatever>, pSrcResource=<whatever>)
```

Or can these arguments only be used positionally, i.e., you'd have to do this:

```python
from d3dshot.dll.d3d import ID3D11DeviceContext
x = ID3D11DeviceContext()
x.CopyResource(<whatever>, <whatever>)
```